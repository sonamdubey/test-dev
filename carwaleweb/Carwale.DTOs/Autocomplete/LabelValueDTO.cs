using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Carwale.DTOs.Autocomplete
{
    public class LabelValueDTO
    {
        [JsonProperty("label")]
        public string Label { get; set; }

        [JsonProperty("value")]
        public string Value { get; set; }
    }
}
