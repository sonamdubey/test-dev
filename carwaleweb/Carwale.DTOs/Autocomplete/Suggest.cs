using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Carwale.DTOs
{

    
    public class BaseAutoCompleteDTO 
    {
        [JsonProperty("displayName")]
        public string DisplayName { get; set; }
    }

    public abstract class PayLoad : BaseAutoCompleteDTO { }
    
    public class Suggest
    {
        [JsonProperty("result")]
        public string Result { get; set; }
        [JsonProperty("payload")]
        public PayLoad Payload { get; set; }
    }

    public class AmpSuggest:Suggest
    {
        [JsonProperty("url")]
        public string Url { get; set; }
    }

    public class CityResultsDTO : PayLoad
    {
        [JsonProperty("cityId")]
        public int Id { get; set; }

        [JsonProperty("cityName")]
        public string Name { get; set; }
		[JsonProperty("cityMaskingName")]
		public string CityMaskingName { get; set; }
		[JsonProperty("isAreaAvailable")]
        public bool IsAreaAvailable { get; set; }

        [JsonProperty("isDuplicate")]
        public bool IsDuplicate { get; set; }
    }

    public class AreaResultsDTO : PayLoad
    {
        [JsonProperty("areaName")]
        public string Name { get; set; }

        [JsonProperty("areaId")]
        public int Id { get; set; }

        [JsonProperty("pinCode")]
        public string PinCode { get; set; }

        [JsonProperty("zoneId")]
        public int ZoneId { get; set; }

        [JsonProperty("zoneName")]
        public string ZoneName { get; set; }

        [JsonProperty("cityId")]
        public int CityId { get; set; }

        [JsonProperty("cityName")]
        public string CityName { get; set; }

        [JsonProperty("cityMaskingName")]
        public string CityMaskingName { get; set; }
    }

    public class CarResultsDTO : PayLoad
    {
        [JsonProperty("makeId")]
        public string MakeId { get; set; }
        [JsonProperty("makeName")]
        public string MakeName { get; set; }
        [JsonProperty("modelId")]
        public string ModelId { get; set; }
        [JsonProperty("modelName")]
        public string ModelName { get; set; }
    }
}
