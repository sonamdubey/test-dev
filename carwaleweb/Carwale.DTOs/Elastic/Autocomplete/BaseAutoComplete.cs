using Newtonsoft.Json;
using System;

namespace Carwale.DTOs.Elastic.Autocomplete
{
    [Serializable]
    public class BaseAutoComplete
    {
        [JsonProperty("displayName")]
        public string DisplayName { get; set; }
    }
}
