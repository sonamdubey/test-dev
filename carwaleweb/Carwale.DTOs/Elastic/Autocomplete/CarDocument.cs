using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Carwale.DTOs.Elastic
{
    [Serializable]
    public class CarDocument
    {
        public ESCarDocument doc { get; set; }
        public ESOperation action { get; set; }
    }
}
