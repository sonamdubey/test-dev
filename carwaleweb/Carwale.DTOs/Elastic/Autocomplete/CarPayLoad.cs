using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Carwale.DTOs.Elastic
{
    [Serializable]
    public class CarPayLoad
    {
        public string CityId { get; set; }
        public string CityName { get; set; }
        public string ColorId { get; set; }
        public string ColorName { get; set; }
        public string MakeId { get; set; }
        public string MakeName { get; set; }
        public string ModelId { get; set; }
        public string ModelName { get; set; }
        public string RootId { get; set; }
        public string RootName { get; set; }
        public string MaskingName { get; set; }
        public string VersionName { get; set; }
        public string OutputName { get; set; }
        public bool IsUpcoming { get; set; }
        public bool IsDiscontinue { get; set; }
        public string VersionId { get; set; }
        public List<CarPayLoad> CarComparisionList { get; set; }
        public string DesktopLink { get; set; }
        public string MobileLink { get; set; }
        public FeaturedCar FeaturedCar { get; set; }
    }
}
