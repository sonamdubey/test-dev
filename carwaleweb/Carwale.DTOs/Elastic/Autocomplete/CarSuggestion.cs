using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Carwale.DTOs.Elastic
{
    [Serializable]
    public class CarSuggestion
    {
        public List<string> input { get; set; }
        public string output { get; set; }
        public CarPayLoad payload { get; set; }
        public Context contexts { get; set; }
        public int weight { get; set; }
        public string suggest { get; set; }
    }
}
