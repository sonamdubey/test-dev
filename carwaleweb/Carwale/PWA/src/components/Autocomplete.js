import React from "react";
import PropTypes from "prop-types";

import AutocompleteNoResult from "./AutocompleteNoResult";

const PropTypeReactComponent = PropTypes.oneOfType([
  PropTypes.func,
  PropTypes.string
]);

const propTypes = {
  // Input properties
  inputProps: PropTypes.object,
  // Input reference
  inputRef: PropTypes.func,
  // Input events
  inputEvents: PropTypes.object,
  // Suggestion list
  list: PropTypes.array,
  // Callback function fired on suggestion list item click
  onSuggestionClick: PropTypes.func,
  // Set focus to input field
  setFocus: PropTypes.bool,
  // Set clear button for input field
  setClearButton: PropTypes.bool,
  // Custom 'No result found' component
  noResultComponent: PropTypeReactComponent,
  // Display no result message
  noResultMessage: PropTypes.string
};

const defaultProps = {
  setClearButton: true,
  noResultComponent: AutocompleteNoResult
};

class Autocomplete extends React.Component {
  constructor(props) {
    super(props);

    this.blurTimer = null;

    this.state = {
      isListActive: false
    };
  }

  componentDidMount() {
    if (this.props.setFocus) {
      this.inputField.focus();
    }
  }

  componentWillUnmount() {
    clearTimeout(this.blurTimer);
  }

  componentWillReceiveProps(nextProps) {
    if (
      this.inputField === document.activeElement &&
      nextProps.inputProps.value &&
      nextProps.list
    ) {
      this.setState({
        isListActive: true
      });
    }
    if (nextProps.setFocus) {
      this.inputField.focus();
    }
  }

  handleInputFocus = () => {
    const { onFocus } = this.props.inputEvents;

    const { value } = this.props.inputProps;

    const { list } = this.props;

    if (value.length && list.length) {
      this.setState({
        isListActive: true
      });
    }

    if (onFocus) {
      onFocus();
    }
  };

  handleInputChange = event => {
    const { onChange } = this.props.inputEvents;

    const value = event.target.value;

    this.setState({
      isListActive: !!value.length && this.props.list.length
    });

    if (onChange) {
      onChange(event);
    }
  };

  handleInputKeyDown = event => {
    const { onKeyDown } = this.props.inputEvents;

    if (onKeyDown) {
      onKeyDown(event);
    }
  };

  handleInputBlur = () => {
    const { onBlur } = this.props.inputEvents;

    this.blurTimer = setTimeout(() => {
      this.setState({
        isListActive: false
      });
    }, 100);

    if (onBlur) {
      onBlur();
    }
  };

  handleInputClear = () => {
    const { onClear } = this.props.inputEvents;

    this.setState({
      isListActive: false
    });

    if (onClear) {
      onClear(this.inputField);
    }
  };

  handleSuggestionMouseDown = (item) => {
    const { onSuggestionClick } = this.props;

    /*
     * According to event order `onBlur` event gets fired before `onClick`,
     * `onClick` callback function was not getting fired on input blur event.
     * Trigger click event with onMouseDown event.
     * https: //stackoverflow.com/questions/17769005/onclick-and-onblur-ordering-issue
     */
    if (onSuggestionClick) {
      onSuggestionClick(item)
    }
  }

	onClick = () => {
		const {
			onClick,
		} = this.props.inputEvents

		if (onClick) {
			onClick()
		}
	}

setReference = ref => {
    const { inputRef } = this.props;
    this.inputField = inputRef ? inputRef(ref) : ref;
  };

  renderSuggestionList = () => {
    const { list } = this.props;

    const styleList = this.state.isListActive ? "show" : "hide";

    if (list) {
      return (
        <ul className={"autocomplete-list " + styleList}>
          {this.suggestionListItem()}
        </ul>
      );
    }
  };

  suggestionListItem = () => {
    const { list, isFetching } = this.props;

    if (list.length) {
      let listElement = list.map((item, index) => {
        return (
          <li key={index} className="autocomplete-list-item">
            <div
              className="autocomplete-list-item__label"
              onMouseDown={this.handleSuggestionMouseDown.bind(this, item)}
            >
              {this.highlighter(item.result)}
            </div>
          </li>
        );
      });

      return listElement;
    } else if (!isFetching) {
      const { noResultComponent: NoResult, noResultMessage } = this.props;

      return <NoResult message={noResultMessage} />;
    }
  };

  highlighter = result => {
    const { value } = this.props.inputProps;

    let escapedPhrase = value.replace(
      /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g,
      "\\$&"
    );
    let tokens = result.split(new RegExp("(" + escapedPhrase + ")", "ig"));
    const regx = new RegExp(escapedPhrase, "i");
    let final = tokens.reduce((acc, token, index) => {
      return acc.concat(
        regx.test(token) ? (
          <strong key={index}>{token}</strong>
        ) : (
          <span key={index}>{token}</span>
        )
      );
    }, []);

    return final;
  };

  render() {
    const { inputProps, setClearButton } = this.props;

    const styleClearTarget = inputProps.value.length ? "show" : "hide";

    return (
      <div className="autocomplete-box">
        <div className="autocomplete-field">
          <input
            type="text"
            ref={this.setReference}
            id={inputProps.id ? inputProps.id : ""}
            placeholder={inputProps.placeholder ? inputProps.placeholder : ""}
            className={inputProps.className ? inputProps.className : ""}
            value={inputProps.value}
            onChange={this.handleInputChange}
            onFocus={this.handleInputFocus}
            onBlur={this.handleInputBlur}
            onKeyDown={this.handleInputKeyDown}
            onClick={this.onClick}
            autoComplete="off"
          />
          {setClearButton && (
            <span
              className={"clear-field-target " + styleClearTarget}
              onClick={this.handleInputClear}
            >
              Clear
            </span>
          )}
        </div>
        {this.renderSuggestionList()}
      </div>
    );
  }
}

Autocomplete.defaultProps = defaultProps;
Autocomplete.propTypes = propTypes;

export default Autocomplete;
