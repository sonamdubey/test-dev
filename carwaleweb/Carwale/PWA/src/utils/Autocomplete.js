const autocomplete = (options, value) => {
  let path = options.url;

  let parameter = '';
  parameter += getValue('term', value);
  parameter += getValue('record', options.resultCount);
  parameter += getValue('cityId', options.cityId);
  parameter += getValue('SourceId', options.source);

  if (parameter) {
    parameter = parameter.slice(0, -1);
    path += '?' + parameter;
  }

  return fetch(path, { method: "GET", headers: { "Accept": "application/json" } }).then(response => response.json())
}

const getValue = (key, value) => {
  if (value) {
    return key + '=' + value + '&';
  }

  return '';
}

export default autocomplete
