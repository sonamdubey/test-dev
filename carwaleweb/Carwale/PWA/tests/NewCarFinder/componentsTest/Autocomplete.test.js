import React from 'react'
import Autocomplete from '../../../src/components/Autocomplete'


describe("Tests for Autocomplete component", () => {
    let props = {}

    let enzymeWrapper

    const Wrapper = () => {
        if(!enzymeWrapper){
            enzymeWrapper = shallow(<Autocomplete { ...props } />)
        }
        return enzymeWrapper
    }

    beforeEach(() => {
        props.inputEvents = {
            onChange: jest.fn(),
            onBlur: jest.fn(),
            onFocus: jest.fn(),
            onClear: jest.fn()
        }
        props.inputProps = {
            id:"",
            placeholder:"",
            className:"",
            value:""
        }
        props.renderSuggestionList = jest.fn()
        props.list = []
    })

    it('should display the value passed', () => {
        props.inputProps = {
            ...props.inputProps,
            value:"Mumbai, Maharashtra",
            placeholder:"Type your city",
            id:"city-input",
            className:"city-field-class"
        }
        const wrapper = Wrapper()
        const inputField = wrapper.find('input').first()
        expect(inputField.props().value).toBe(props.inputProps.value)
        expect(inputField.props().placeholder).toBe(props.inputProps.placeholder)
        expect(inputField.props().id).toBe(props.inputProps.id)
        expect(inputField.props().className).toBe(props.inputProps.className)
        expect(wrapper).toMatchSnapshot()
    })

    it('should call respective methods when events are triggered', () => {
        const wrapper = Wrapper()

        const inputField = wrapper.find('input')

        const instanceEventHandlers = wrapper.instance().props.inputEvents

        expect(wrapper.instance().props.renderSuggestionList).toBeCalledWith(props.list)

        inputField.simulate('focus')
        expect(instanceEventHandlers.onFocus).toBeCalled()

        inputField.simulate('blur')
        expect(instanceEventHandlers.onBlur).toBeCalled()

        inputField.simulate('change', { target: { value: 'mum'}})
        expect(instanceEventHandlers.onChange).toBeCalled()

        wrapper.find('.clear-field-target').simulate('click')
        expect(instanceEventHandlers.onClear).toBeCalled()

    })
})
