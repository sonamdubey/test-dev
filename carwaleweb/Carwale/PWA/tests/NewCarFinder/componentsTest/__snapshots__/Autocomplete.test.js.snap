// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Autocomplete component should display the value passed 1`] = `
<div
  className="autocomplete-box"
>
  <div
    className="autocomplete-field"
  >
    <input
      className="city-field-class"
      id="city-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Type your city"
      type="text"
      value="Mumbai, Maharashtra"
    />
    <span
      className="clear-field-target show"
      onClick={[Function]}
    >
      Clear
    </span>
  </div>
</div>
`;
